src/service-providers/mailchimp/ProviderSidebar.js

```diff
@@ -3,75 +3,71 @@
  */
 import { __ } from '@wordpress/i18n';
 import { Fragment, useEffect, useState } from '@wordpress/element';
-import {
-	ExternalLink,
-	SelectControl,
-	Spinner,
-	Notice,
-	FormTokenField,
-} from '@wordpress/components';
+import { ExternalLink, SelectControl, Spinner, Notice } from '@wordpress/components';
 
 /**
  * Internal dependencies
  */
-import { getListInterestsSettings, getListTags, getTagNames, getTagIds } from './utils';
+import { getListInterestsSettings } from './utils';
 
 const SegmentsSelection = ( {
 	onUpdate,
 	inFlight,
-	chosenInterestId,
+	chosenTarget,
 	availableInterests,
-	chosenTags,
-	availableTags,
+	availableSegments,
 } ) => {
-	const [ segmentsData, setSegmentsData ] = useState( {
-		interest_id: chosenInterestId,
-		tag_ids: chosenTags,
-	} );
-	const updateSegmentsData = data => setSegmentsData( { ...segmentsData, ...data } );
-
-	// Update with real data after local (optimistic) update.
-	useEffect(() => {
-		setSegmentsData( {
-			interest_id: chosenInterestId,
-			tag_ids: chosenTags,
-		} );
-	}, [ chosenInterestId, chosenTags.join() ]);
+	const [ targetId, setTargetId ] = useState( chosenTarget || '' );
 
 	const [ isInitial, setIsInitial ] = useState( true );
 	useEffect(() => {
 		if ( ! isInitial ) {
-			onUpdate( segmentsData );
+			onUpdate( targetId );
 		}
 		setIsInitial( false );
-	}, [ segmentsData.interest_id, segmentsData.tag_ids.join() ]);
+	}, [ targetId ]);
+
+	let options = [];
+
+	if ( availableInterests.length > 0 ) {
+		options = options.concat( [
+			{
+				label: __( 'Group', 'newspack-newsletters' ),
+				value: 'groups',
+				disabled: true,
+			},
+			...availableInterests,
+		] );
+	}
+
+	if ( availableSegments.length > 0 ) {
+		options = options.concat( [
+			{
+				label: __( 'Segment or tag', 'newspack-newsletters' ),
+				value: 'segments',
+				disabled: true,
+			},
+			...availableSegments.map( segment => ( {
+				label: ` - ${ segment.name }`,
+				value: segment.id,
+			} ) ),
+		] );
+	}
 
 	return (
 		<Fragment>
-			{ availableInterests.length ? (
+			{ options.length ? (
 				<SelectControl
-					label={ __( 'Groups', 'newspack-newsletters' ) }
-					value={ segmentsData.interest_id }
+					label={ __( 'Group, segment or tag', 'newspack-newsletters' ) }
```

Comment:
Oxford-comma nitpick ;) : 
```suggestion
					label={ __( 'Group, segment, or tag', 'newspack-newsletters' ) }
```